core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/BlobSidecar.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.core.service.http.handlers.signing.eth2;
+
+import tech.pegasys.teku.api.schema.KZGCommitment;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.SpecVersion;
+import tech.pegasys.teku.spec.datastructures.blobs.versions.deneb.BlindedBlobSidecarSchema;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.tuweni.bytes.Bytes32;
+
+/**
+ * Json representation of BlindedBlobSidecar and BlobSidecar. The blob_root is the "hash tree root"
+ * of BlobSidecar's blob
+ *
+ * @param blockRoot Block root
+ * @param index Index
+ * @param slot Slot
+ * @param blockParentRoot Block Parent Root
+ * @param proposerIndex Proposer Index
+ * @param blobRoot Blob Root.
+ * @param kzgCommitment KZG Commitment
+ * @param kzgProof KZG Proof
+ */
+public record BlobSidecar(
+    @JsonProperty("block_root") Bytes32 blockRoot,
+    @JsonProperty("index") UInt64 index,
+    @JsonProperty("slot") UInt64 slot,
+    @JsonProperty("block_parent_root") Bytes32 blockParentRoot,
+    @JsonProperty("proposer_index") UInt64 proposerIndex,
+    @JsonProperty("blob_root") Bytes32 blobRoot,
+    @JsonProperty("kzg_commitment") KZGCommitment kzgCommitment,
+    @JsonProperty("kzg_proof") KZGProof kzgProof) {
+
+  public tech.pegasys.teku.spec.datastructures.blobs.versions.deneb.BlindedBlobSidecar
+      asInternalBlindedBlobSidecar(final SpecVersion spec) {
+    final BlindedBlobSidecarSchema blindedBlobSidecarSchema =
```

Comment:
Should this be BlobSidecar type instead now?

Reply:
this is internal (Teku) class.